test/stmt/if_while_var.swift

```diff
@@ -51,7 +51,9 @@ guard let _ = nonOptionalEnum() else { fatalError() } // expected-error{{initial
 let optional: String? = nil
 if case let optional? { _ = optional } // expected-error{{variable binding in a condition requires an initializer}}
 if case let .some(optional) { _ = optional } // expected-error{{variable binding in a condition requires an initializer}}
+                                             // expected-error@-1 {{type of expression is ambiguous without more context}}
 if case .some(let optional) { _ = optional } // expected-error{{variable binding in a condition requires an initializer}}
+                                             // expected-error@-1 {{type of expression is ambiguous without more context}}
```

Comment:
Why did this start producing `type of expression is ambiguous without more context`?

Reply:
I just investigated this failure and we had free type variables left in the solution when solving it. Changing the type variable returned by `getTypeForPattern` to allow binding to a hole fixes the issue. (CSGen.cpp:2330).

Reply:
Perfect!